### PR TITLE
dns/server: allow configurable use of unbound

### PR DESCRIPTION
**Pull request by tynes**
This PR adds a new config option `rs-no-unbound` that is expected to be a boolean. If set to true, it will not use `unbound` and instead use the node.js based `RecursiveResolver`.

Since `initNode` sets listeners on `this.hns`, it is moved to after the call to `initOptions`, since `initOptions` can will overwrite `this.hns` if `noUnbound` is set to true.

### CODE REVIEW

**Comment by tynes** [comment]
@boymanjor This PR is meant to allow for deterministic testing of the DNS server. It is possible that the `unbound` bindings differ slightly from the node.js DNS server implementation and this config option would allow testing of both configurations without the need to delete the `unbound` header files and rebuild `hsd`.

**Comment by boymanjor** [comment]
Minor nit: the commit prefix should use directory structure, i.e. `dns/server`.

**Comment by tynes** [comment]
Updated commit message to use directory structure

File: lib/dns/server.js [hunk 1]
```diff
@@ -535,6 +536,16 @@ class RecursiveServer extends DNSServer {
       this.stubPort = options.stubPort;
     }
 
+    if (options.noUnbound === true) {
```
Comment (boymanjor) [inline lib/dns/server.js]: I know we discussed this out of band, but now I think we should fully validate `noUnbound` here, i.e.:
```js
if (options.noUnbound != null) {
  assert(typeof options.noUnbound === 'boolean')
  if (options.noUnbound) {
    ...
  }
}
```

The motivation is to protect against one-off imports of this module that do not use `bcfg` to validate certain properties of the options hash.

Comment (tynes) [reply lib/dns/server.js]: Good call. Pushed up a fix for this.

**Pull request merged**